nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/grok/GrokRecordReader.java

```diff
@@ -0,0 +1,318 @@
+        // the next call to nextRecord()) or we will attach it to the previously read record.
+        String stackTrace = null;
+        final StringBuilder toAppend = new StringBuilder();
+        while ((nextLine = reader.readLine()) != null) {
+            final Match nextLineMatch = grok.match(nextLine);
+            nextLineMatch.captures();
+            final Map<String, Object> nextValueMap = nextLineMatch.toMap();
+            if (nextValueMap.isEmpty()) {
+                // next line did not match. Check if it indicates a Stack Trace. If so, read until
+                // the stack trace ends. Otherwise, append the next line to the last field in the record.
+                if (isStartOfStackTrace(nextLine)) {
+                    stackTrace = readStackTrace(nextLine);
+                    break;
+                } else {
+                    toAppend.append("\n").append(nextLine);
+                }
+            } else {
+                // The next line matched our pattern.
+                break;
+            }
+        }
+
+        try {
+            final List<DataType> fieldTypes = schema.getDataTypes();
+            final Map<String, Object> values = new HashMap<>(fieldTypes.size());
+
+            for (final String fieldName : schema.getFieldNames()) {
+                final Object value = valueMap.get(fieldName);
+                if (value == null) {
+                    values.put(fieldName, null);
+                    continue;
+                }
+
+                final DataType fieldType = schema.getDataType(fieldName).orElse(null);
+                final Object converted = convert(fieldType, value.toString());
+                values.put(fieldName, converted);
+            }
+
+            final String lastFieldBeforeStackTrace = schema.getFieldNames().get(schema.getFieldCount() - 2);
+            if (toAppend.length() > 0) {
+                final Object existingValue = values.get(lastFieldBeforeStackTrace);
+                final String updatedValue = existingValue == null ? toAppend.toString() : existingValue + toAppend.toString();
+                values.put(lastFieldBeforeStackTrace, updatedValue);
+            }
+
+            values.put(STACK_TRACE_COLUMN_NAME, stackTrace);
+
+            return new MapRecord(schema, values);
+        } catch (final Exception e) {
+            throw new MalformedRecordException("Found invalid log record and will skip it. Record: " + line, e);
+        }
+    }
+
+
+    private boolean isStartOfStackTrace(final String line) {
+        if (line == null) {
+            return false;
+        }
+
+        // Stack Traces are generally of the form:
+        // java.lang.IllegalArgumentException: My message
+        //   at org.apache.nifi.MyClass.myMethod(MyClass.java:48)
+        //   at java.lang.Thread.run(Thread.java:745) [na:1.8.0_60]
+        // Caused by: java.net.SocketTimeoutException: null
+        //   ... 13 common frames omitted
```

Comment:
I did. Just because when debugging I had to generate stack traces a few times. So I just added a comment that included this so that it'll be easier for debugging later